Guessing the test fail is real.
```
======================================================================
ERROR: custom.icds_reports.tests.reports.test_awc_reports:TestAWCReport.test_awc_report_lactating_fifth_record
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/custom/icds_reports/tests/reports/test_awc_reports.py", line 2686, in test_awc_report_lactating_fifth_record
    data['data'][4],
IndexError: list index out of range
----------------------------------------------------------------------
```

--- a/custom/icds_reports/reports/awc_reports.py
+++ b/custom/icds_reports/reports/awc_reports.py
@@ -1151,7 +1151,7 @@ def get_awc_report_pregnant(start, length, order, reversed_order, awc_id):
 
 @icds_quickcache(['case_id', 'awc_id'], timeout=30 * 60)
 def get_pregnant_details(case_id, awc_id):
-    ten_months_ago = datetime.utcnow() - relativedelta(months=10, day=1)
+    ten_months_ago = date.today() - relativedelta(months=10, day=1)
     data = CcsRecordMonthlyView.objects.filter(
         case_id=case_id,
         awc_id=awc_id,

--- a/custom/icds_reports/tests/reports/test_awc_reports.py
+++ b/custom/icds_reports/tests/reports/test_awc_reports.py
@@ -2492,7 +2492,7 @@ class TestAWCReport(TestCase):
         )
 
     def test_awc_report_pregnant_first_record(self):
-        with mock.patch('custom.icds_reports.reports.awc_reports.date', FirstDayOfMayDate):
+        with mock.patch('custom.icds_reports.reports.awc_reports.date', SecondDayOfMay):
             data = get_awc_report_pregnant(
                 start=0,
                 length=10,
@@ -2525,7 +2525,7 @@ class TestAWCReport(TestCase):
             )
 
     def test_pregnant_details_first_record_first_trimester(self):
-        with mock.patch('custom.icds_reports.reports.awc_reports.date', FirstDayOfMay):
+        with mock.patch('custom.icds_reports.reports.awc_reports.date', SecondDayOfMay):
             data = get_pregnant_details(
                 case_id='7313c174-6b63-457c-a734-6eed0a2b2ac6',
                 awc_id='a15'
@@ -2536,7 +2536,7 @@ class TestAWCReport(TestCase):
             )
 
     def test_pregnant_details_first_record_second_trimester(self):
-        with mock.patch('custom.icds_reports.reports.awc_reports.date', FirstDayOfMay):
+        with mock.patch('custom.icds_reports.reports.awc_reports.date', SecondDayOfMay):
             data = get_pregnant_details(
                 case_id='7313c174-6b63-457c-a734-6eed0a2b2ac6',
                 awc_id='a15'
@@ -2567,7 +2567,7 @@ class TestAWCReport(TestCase):
             )
 
     def test_pregnant_details_first_record_third_trimester(self):
-        with mock.patch('custom.icds_reports.reports.awc_reports.date', FirstDayOfMay):
+        with mock.patch('custom.icds_reports.reports.awc_reports.date', SecondDayOfMay):
             data = get_pregnant_details(
                 case_id='7313c174-6b63-457c-a734-6eed0a2b2ac6',
                 awc_id='a15'

--- a/custom/icds_reports/tests/reports/test_awc_reports.py
+++ b/custom/icds_reports/tests/reports/test_awc_reports.py
@@ -20,10 +20,10 @@ from custom.icds_reports.messages import new_born_with_low_weight_help_text, was
 from six.moves import filter
 
 
-class FirstDayOfMay(datetime.datetime):
+class FirstDayOfMay(date):
     @classmethod
-    def utcnow(cls):
-        return datetime.datetime(2017, 5, 1)
+    def today(cls):
+        return date(2017, 5, 1)
 
 
 class FirstDayOfMayDate(date):
@@ -34,8 +34,8 @@ class FirstDayOfMayDate(date):
 
 class SecondDayOfMay(date):
     @classmethod
-    def utcnow(cls):
-        return datetime.datetime(2017, 5, 2)
+    def today(cls):
+        return date(2017, 5, 2)
 
 
 class TestAWCReport(TestCase):
@@ -2525,7 +2525,7 @@ class TestAWCReport(TestCase):
             )
 
     def test_pregnant_details_first_record_first_trimester(self):
-        with mock.patch('custom.icds_reports.reports.awc_reports.datetime', FirstDayOfMay):
+        with mock.patch('custom.icds_reports.reports.awc_reports.date', FirstDayOfMay):
             data = get_pregnant_details(
                 case_id='7313c174-6b63-457c-a734-6eed0a2b2ac6',
                 awc_id='a15'
@@ -2536,7 +2536,7 @@ class TestAWCReport(TestCase):
             )
 
     def test_pregnant_details_first_record_second_trimester(self):
-        with mock.patch('custom.icds_reports.reports.awc_reports.datetime', FirstDayOfMay):
+        with mock.patch('custom.icds_reports.reports.awc_reports.date', FirstDayOfMay):
             data = get_pregnant_details(
                 case_id='7313c174-6b63-457c-a734-6eed0a2b2ac6',
                 awc_id='a15'
@@ -2567,7 +2567,7 @@ class TestAWCReport(TestCase):
             )
 
     def test_pregnant_details_first_record_third_trimester(self):
-        with mock.patch('custom.icds_reports.reports.awc_reports.datetime', FirstDayOfMay):
+        with mock.patch('custom.icds_reports.reports.awc_reports.date', FirstDayOfMay):
             data = get_pregnant_details(
                 case_id='7313c174-6b63-457c-a734-6eed0a2b2ac6',
                 awc_id='a15'
@@ -2578,7 +2578,7 @@ class TestAWCReport(TestCase):
             )
 
     def test_awc_report_lactating_first_record(self):
-        with mock.patch('custom.icds_reports.reports.awc_reports.datetime', SecondDayOfMay):
+        with mock.patch('custom.icds_reports.reports.awc_reports.date', SecondDayOfMay):
             data = get_awc_report_lactating(
                 start=0,
                 length=10,
@@ -2602,7 +2602,7 @@ class TestAWCReport(TestCase):
             )
 
     def test_awc_report_lactating_second_record(self):
-        with mock.patch('custom.icds_reports.reports.awc_reports.datetime', SecondDayOfMay):
+        with mock.patch('custom.icds_reports.reports.awc_reports.date', SecondDayOfMay):
             data = get_awc_report_lactating(
                 start=0,
                 length=10,
@@ -2626,7 +2626,7 @@ class TestAWCReport(TestCase):
             )
 
     def test_awc_report_lactating_third_record(self):
-        with mock.patch('custom.icds_reports.reports.awc_reports.datetime', SecondDayOfMay):
+        with mock.patch('custom.icds_reports.reports.awc_reports.date', SecondDayOfMay):
             data = get_awc_report_lactating(
                 start=0,
                 length=10,
@@ -2650,7 +2650,7 @@ class TestAWCReport(TestCase):
             )
 
     def test_awc_report_lactating_forth_record(self):
-        with mock.patch('custom.icds_reports.reports.awc_reports.datetime', SecondDayOfMay):
+        with mock.patch('custom.icds_reports.reports.awc_reports.date', SecondDayOfMay):
             data = get_awc_report_lactating(
                 start=0,
                 length=10,
@@ -2674,7 +2674,7 @@ class TestAWCReport(TestCase):
             )
 
     def test_awc_report_lactating_fifth_record(self):
-        with mock.patch('custom.icds_reports.reports.awc_reports.datetime', SecondDayOfMay):
+        with mock.patch('custom.icds_reports.reports.awc_reports.date', SecondDayOfMay):
             data = get_awc_report_lactating(
                 start=0,
                 length=10,
@@ -2698,7 +2698,7 @@ class TestAWCReport(TestCase):
             )
 
     def test_awc_report_lactating_sixth_record(self):
-        with mock.patch('custom.icds_reports.reports.awc_reports.datetime', SecondDayOfMay):
+        with mock.patch('custom.icds_reports.reports.awc_reports.date', SecondDayOfMay):
             data = get_awc_report_lactating(
                 start=0,
                 length=10,
@@ -2722,7 +2722,7 @@ class TestAWCReport(TestCase):
             )
 
     def test_awc_report_lactating_seventh_record(self):
-        with mock.patch('custom.icds_reports.reports.awc_reports.datetime', SecondDayOfMay):
+        with mock.patch('custom.icds_reports.reports.awc_reports.date', SecondDayOfMay):
             data = get_awc_report_lactating(
                 start=0,
                 length=10,
@@ -2747,7 +2747,7 @@ class TestAWCReport(TestCase):
             )
 
     def test_awc_report_lactating_on_first_of_month(self):
-        with mock.patch('custom.icds_reports.reports.awc_reports.datetime', FirstDayOfMay):
+        with mock.patch('custom.icds_reports.reports.awc_reports.date', FirstDayOfMay):
             data = get_awc_report_lactating(
                 start=0,
                 length=7,

